### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by sudeeprp** [clean-code-craft-tcq-2/tdd-buckets-Nagamuni-Ratakonda#1]
@Nagamuni-Ratakonda I am commenting in a new issue since there was no Pull request

`SortCurrentRanges` - Why not use built-in [qsort](https://www.tutorialspoint.com/c_standard_library/c_function_qsort.htm)
That is something that the complexity limit tells you... when you are doing something complex, is it unique to your solution, or can you pick off-the-shelf?

`CheckForPeriodicSetRanges` prints the heading of the csv, but it is called in a loop from `CaptureChargingCurrentRanges`. Will the heading be printed multiple times? As an exercise - try to design a test to validate the csv and fail multiple headings, then fix it.

**Comment by Nagamuni-Ratakonda** [comment]
1. Used built-in qsort function for sorting the array
2. Heading is moved out of for loop. Now heading will be printed only once.
